src/dndbeyond_discord.pyj

```diff
@@ -1,10 +1,18 @@
 from constants import DISCORD_BOT_API_URL
+from settings import WhisperType
 
 v'async'
 def postToDiscord(secret, request, title, source, attributes, description, attack_rolls, roll_info, damage_rolls, total_damages, open):
     secret = (secret or "").trim()
     if not secret:
         return
+
+    if request.character.type == "Monster":
+        if request.whisper_monster in [WhisperType.YES, WhisperType.HIDE_NAMES]:

```

Comment:
If the whisper type is HIDE_NAMES, then it should just hide it. You don't need to check if it's a monster and check againster the whisper_monster value.